components/service/glean/src/main/java/mozilla/components/service/glean/EventMetricType.kt

```diff
@@ -109,4 +111,31 @@ data class EventMetricType(
             )
         }
     }
+
+    /**
+     * Tests whether a value is stored for the metric for testing purposes only
+     *
+     * @return true if metric value exists, otherwise false
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    fun testHasValue(
+        pingName: String =
+            if (sendInPings[0] == "default") "metrics"
+            else sendInPings[0]): Boolean {
+        return EventsStorageEngine.getSnapshot(pingName,false) != null
```

Comment:
I think this will return `true` if there are *any* events in the store, not just events with the name of this specific event, which is probably confusing.  I think it needs to do a `filter` (or something) on events with the metric's name and category.

Reply:
Gotcha, thanks!  I'm fighting ktlint and detekt at the moment, I'll fix this as soon as I appease the style gods.

Reply:
Filtered now to check if both the ping snapshot is there (this is what it was doing), as well as if the `category.name` (or just `name` if no `category`) exists

Reply:
refactored to use `.any { ... }`